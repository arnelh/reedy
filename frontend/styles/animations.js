import { css } from "glamor";

export const spinAnimation = css.keyframes({
  "0%": { transform: "rotate(0deg)" },
  "100%": { transform: "rotate(360deg)" },
});

export const spin = () => {
  return { animation: `${spinAnimation} 0.4s infinite linear` };
};

export const opacityChange = css.keyframes({
  "0%": { opacity: "0" },
  "60%": { opacity: "1" },
  "100%": { opacity: "0" },
});

export const opacity = () => (window.opacityChange = opacityChange);
