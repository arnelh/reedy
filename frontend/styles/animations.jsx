import { css } from "glamor";

export const spin = props => {
  const spinAnimation = css.keyframes({
    "0%": { transform: "rotate(0deg)" },
    "100%": { transform: "rotate(360deg)" },
  });

  return { animation: `${spinAnimation} 0.4s infinite linear` };
};
