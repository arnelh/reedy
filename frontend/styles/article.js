import glamorous from "glamorous";
import { SOFT_BOX_SHADOW, DARKER } from "./theme";
const ARTICLE_ITEM_HEIGHT = 150;

export const StyledArticleListWrapper = glamorous.div({
  maxWidth: 740,
  paddingTop: 20,
});

export const StyledArticleItemWrapper = glamorous.div({
  display: "flex",
  margin: 10,
  justifyContent: "flex-start",
  height: ARTICLE_ITEM_HEIGHT,
  position: "relative",
});

export const ArticleSmallImg = glamorous.img({
  maxHeight: ARTICLE_ITEM_HEIGHT - 20,
  maxWidth: ARTICLE_ITEM_HEIGHT - 20,
});

export const ImgWrapper = glamorous.div(
  {
    display: "flex",
    alignItems: "flex-start",
  },
  ({ hasImage }) => ({
    height: hasImage ? ARTICLE_ITEM_HEIGHT : 0,
    width: hasImage ? ARTICLE_ITEM_HEIGHT : 0,
  })
);

export const StyledArticleSnippetWrapper = glamorous.div(
  {
    display: "flex",
    flexDirection: "column",
  },
  ({ hasImage }) => ({
    width: hasImage ? `calc(90% - ${ARTICLE_ITEM_HEIGHT}px)` : "90%",
  })
);

export const ArticleMiniTitle = glamorous.h4({
  width: "100%",
  fontWeight: 500,
});

export const ArticleSnippet = glamorous.p({
  fontFamily: "Oxygen",
  fontSize: 14,
  color: "#626262",
  margin: 5,
  overflow: "hidden",
  position: "relative",
  maxHeight: 14 * 5,
  textAlign: "justify",
  marginRight: "-1em",
  paddingPight: "1em",
  ":before": {
    content: "...",
    position: "absolute",
    right: 0,
    bottom: 0,
  },
  ":after": {
    content: "close-quote",
    position: "absolute",
    right: 0,
    width: "1em",
    height: "1em",
    marginTop: "0.2em",
    background: "white",
  },
});

export const StyledArticleModal = glamorous.div({
  position: "fixed",
  top: 0,
  left: 0,
  width: "100vw",
  height: "100vh",
  backgroundColor: "rgba(0, 0, 0, 1)",
  cursor: "pointer",
  zIndex: 5,
});

export const ArticleDetailWrapper = glamorous.div({
  top: 0,
  right: 0,
  boxSizing: "border-box",
  width: "70%",
  height: "100%",
  backgroundColor: "#fafafa",
  boxShadow: "18px 18px 90px 2px rgba(0,0,0,0.5)",
  cursor: "auto",
  position: "absolute",
  overflow: "auto",
  zIndex: 6,
});

export const ArticleDetailContent = glamorous.div({
  backgroundColor: "#fff",
  boxShadow: SOFT_BOX_SHADOW,
  margin: "0 30px",
  padding: 40,
  marginTop: 55,
});

export const StyledArticleDetailNav = glamorous.div({
  height: "100%",
  background: "#fff",
  zIndex: 3,
});

export const StyledActionBar = glamorous.div({
  height: 50,
  width: 50,
  position: "absolute",
  right: 0,
  display: "flex",
  justifyContent: "center",
});

export const StyledBookmark = glamorous.i({
  fontSize: 20,
  color: DARKER,
});

export const StyledNavActionBar = glamorous.div({
  height: 100,
});

export const DetailNavLeftWrapper = glamorous.div({
  position: "absolute",
  height: "100%",
  display: "flex",
  justifyContent: "space-between",
  top: 7,
  width: 100,
  left: 30,
});

export const DetailNavIcon = glamorous.i(
  {
    textAlign: "center",
    fontSize: 20,
    width: 20,
    height: 20,
    borderRadius: 3,
    transition: "all 100ms",
    ":hover": {
      transition: "all 100ms",
      color: DARKER,
      cursor: "pointer",
      border: `1px solid ${DARKER}`,
    },
  },
  props => ({
    color: props.saved ? DARKER : "#797979",
    border: props.saved ? `1px solid ${DARKER}` : "1px solid #e7e7e7",
  })
);
