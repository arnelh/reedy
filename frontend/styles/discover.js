import glamorous from "glamorous";
import { SOFT_BOX_SHADOW, SHADOW_BORDER } from "./theme";

export const StyledDiscoverItem = glamorous.div({
  height: 200,
  width: 200,
  boxShadow: SOFT_BOX_SHADOW,
  border: SHADOW_BORDER,
  display: "flex",
  flexDirection: "column",
  alignItems: "center",
  flexWrap: "wrap",
  margin: 10,
  transition: "all 0.3s cubic-bezier(.25,.8,.25,1)",
  backgroundColor: "#fff",
  ":hover": {
    boxShadow: "0 10px 28px rgba(0,0,0,0.25), 0 6px 10px rgba(0,0,0,0.22)",
    transform: "translatey(-5px)",
  },
});

export const FeedTitle = glamorous.div({
  width: "100%",
  textAlign: "center",
});

export const FeedImg = glamorous.img({
  maxHeight: "100%",
  maxWidth: "100%",
});

export const FeedImgContainer = glamorous.div({
  width: 60,
  height: 60,
  boxShadow: SOFT_BOX_SHADOW,
  border: SHADOW_BORDER,
  display: "flex",
  justifyContent: "center",
  alignItems: "center",
});

export const StyledDiscoverWrapper = glamorous.div({
  padding: "105px 0",
  maxWidth: 960,
  display: "flex",
  flexWrap: "wrap",
  justifyContent: "center",
  alignContent: "flex-start",
});

export const DiscoverItemWrapper = glamorous.div({
  width: 180,
  height: 140,
  display: "flex",
  flexDirection: "column",
  justifyContent: "space-around",
  alignItems: "center",
});
