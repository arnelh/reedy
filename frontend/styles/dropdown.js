import glamorous from "glamorous";
import { RED, BASE, DARKER, Button, NAV_HEIGHT } from "./theme";

const DROPDOWN_HEIGHT = 100;
const PADDING = 5;

export const StyledDropdown = glamorous.div(
  {
    position: "absolute",
    width: "140px",
    height: DROPDOWN_HEIGHT,
    top: `calc(${NAV_HEIGHT}px + 3px)`,
    right: 0,
    backgroundColor: "#fff",
    border: "1px solid #dddede",
    borderRadius: "4px",
    fontFamily: "Oxygen",
    boxShadow: "0 3px 12px rgba(27,31,35,0.15)",
    "&::before": {
      content: "close-quote",
      position: "absolute",
      display: "inline-block",
      top: "-16px",
      right: "9px",
      left: "auto",
      border: "8px solid transparent",
      borderBottomColor: "#dddede",
    },
    "&::after": {
      content: "close-quote",
      border: "7px solid transparent",
      borderBottomColor: "#fff",
      position: "absolute",
      display: "inline-block",
      top: "-14px",
      right: "10px",
      left: "auto",
    },
  },
  props => ({
    display: props.active ? "flex" : "none",
  })
);

export const DropdownContent = glamorous.div({
  width: "100%",
  height: "100%",
  display: "flex",
  justifyContent: "center",
  flexDirection: "column",
  position: "relative",
});

export const DropdownButton = glamorous(Button)({
  height: 45,
  width: "100%",
  color: "#111",
  textAlign: "left",
  fontSize: "14px",
  paddingLeft: "10px",
  fontWeight: 500,
  backgroundColor: "#fff",
  ":hover": {
    color: "#eee",
  },
});

export const DropdownEmptyItem = glamorous.div({
  lineHeight: 1.4,
  boxSizing: "border-box",
  display: "flex",
  paddingTop: PADDING,
  paddingLeft: "10px",
  flexDirection: "column",
  justifyContent: "center",
  fontSize: "14px",
  height: 50,
});

export const Line = glamorous.div({
  height: PADDING,
  borderBottom: "1px solid #e7e7e7",
});

export const Bold = glamorous.p({
  fontWeight: 500,
});

export const StyledFeedDropdown = glamorous(StyledDropdown)({
  height: "auto",
  zIndex: 2,
  top: 70,
  right: 10,
  width: 200,
});

export const StyledFeedButtons = glamorous(DropdownButton)(
  {
    backgroundColor: "#fff",
    fontWeight: 300,
    fontFamily: "Roboto",
    display: "flex",
    alignItems: "center",
    textAlign: "center",
    position: "relative",
    ":hover": {
      backgroundColor: "#f3f3f3",
      color: "#454545",
    },
  },
  props => ({
    color: props.followed ? DARKER : "#454545",
    ":hover": {
      color: props.followed ? DARKER : "#454545",
    },
  })
);

export const RSSSquare = glamorous.i({
  fontSize: 16,
  marginRight: 5,
});

export const IndicatorIcons = glamorous(RSSSquare)(
  {
    position: "absolute",
    right: 10,
    top: "30%",
  },
  props => ({
    color: props.delete ? RED : "auto",
  })
);

export const StyledCreateCollectionButton = glamorous(Button)({
  fontWeight: 300,
  fontFamily: "Roboto",
  width: "100%",
  height: 45,
  position: "relative",
  fontSize: 12,
  padding: 0,
  backgroundColor: BASE,
  textAlign: "center",
  color: "#f3f3f3",
});

export const StyledCreateInput = glamorous.input({
  boxSizing: "border-box",
  width: "100%",
  height: 40,
  marginTop: 10,
  borderRadius: 5,
  paddingLeft: 10,
  outline: "none",
  border: `1px solid ${BASE}`,
  fontSize: 14,
});

export const DropdownActionButton = glamorous(StyledCreateCollectionButton)({
  height: 40,
  width: "48%",
  borderRadius: 5,
  padding: 0,
  backgroundColor: "#fff",
  textAlign: "center",
});

export const StyledDropdownSubmitButton = glamorous(DropdownActionButton)({
  border: `1px solid ${BASE}`,
  color: BASE,
  ":hover": {
    backgroundColor: BASE,
    color: "#f3f3f3",
  },
});

export const StyledDropdownCancelButton = glamorous(DropdownActionButton)({
  border: `1px solid ${RED}`,
  color: RED,
  ":hover": {
    backgroundColor: RED,
    color: "#f3f3f3",
  },
});

export const ButtonContainer = glamorous.div({
  width: "100%",
  marginTop: 10,
  display: "flex",
  alignItems: "center",
  justifyContent: "space-between",
});

export const StyledCreateForm = glamorous.form({
  paddingBottom: 10,
  width: "90%",
  display: "flex",
  flexDirection: "column",
  justifyContent: "space-between",
  alignSelf: "center",
});
