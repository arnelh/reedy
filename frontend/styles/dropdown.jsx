import glamorous from "glamorous";
import { Button, NAV_HEIGHT } from "./theme";

export const StyledDropdown = glamorous.div(
  {
    position: "absolute",
    width: "200px",
    height: "230px",
    top: `calc(${NAV_HEIGHT} + 3px)`,
    right: 0,
    backgroundColor: "#fff",
    border: "1px solid #dddede",
    borderRadius: "4px",
    boxShadow: "0 3px 12px rgba(27,31,35,0.15)",
    "&::before": {
      content: "close-quote",
      position: "absolute",
      display: "inline-block",
      top: "-16px",
      right: "9px",
      left: "auto",
      border: "8px solid transparent",
      borderBottomColor: "#dddede",
    },
    "&::after": {
      content: "close-quote",
      border: "7px solid transparent",
      borderBottomColor: "#fff",
      position: "absolute",
      display: "inline-block",
      top: "-14px",
      right: "10px",
      left: "auto",
    },
  },
  props => ({
    display: props.active ? "flex" : "none",
  })
);

export const DropdownList = glamorous.ul({
  width: "100%",
  height: "100%",
  display: "grid",
  listStyle: "none",
});

export const DropdownButton = glamorous(Button)({
  width: "100%",
  height: "100%",
  fontSize: "16px",
});
