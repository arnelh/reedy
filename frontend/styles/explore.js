import glamorous from "glamorous";
import { NAV_BOX_SHADOW, SHADOW_BORDER } from "./theme";

export const StyledExploreItem = glamorous.div({
  height: 200,
  width: 200,
  boxShadow: NAV_BOX_SHADOW,
  border: SHADOW_BORDER,
  display: "flex",
  flexDirection: "column",
  alignItems: "center",
  flexWrap: "wrap",
  margin: 10,
  transition: "all 0.3s cubic-bezier(.25,.8,.25,1)",
  ":hover": {
    boxShadow: "0 10px 28px rgba(0,0,0,0.25), 0 6px 10px rgba(0,0,0,0.22)",
    transform: "translatey(-5px)",
  },
});

export const FeedTitle = glamorous.div({
  width: "100%",
  textAlign: "center",
});

export const FeedImg = glamorous.img({
  maxHeight: "100%",
  maxWidth: "100%",
});

export const FeedImgContainer = glamorous.div({
  width: 60,
  height: 60,
  boxShadow: NAV_BOX_SHADOW,
  border: SHADOW_BORDER,
  display: "flex",
  justifyContent: "center",
  alignItems: "center",
});

export const StyledExploreWrapper = glamorous.div({
  margin: 50,
  maxWidth: 960,
  display: "flex",
  flexWrap: "wrap",
  justifyContent: "center",
  alignContent: "flex-start",
});

export const ExploreItemWrapper = glamorous.div({
  width: 180,
  height: 140,
  display: "flex",
  flexDirection: "column",
  justifyContent: "space-around",
  alignItems: "center",
});
