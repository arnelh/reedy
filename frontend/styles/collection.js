import glamorous from "glamorous";
import { FeedImgContainer } from "./discover";
import { HALF_TRANSPARENT, SLIGHT_WHITE, SIDE_NAV_WIDTH } from "./theme";

export const StyledCollectionWrapper = glamorous.div({
  width: SIDE_NAV_WIDTH,
  boxSizing: "border-box",
  height: "calc(100% - 60px)",
  overflow: "auto",
  paddingTop: 20,
});

export const StyledItemBox = glamorous.div({
  width: SIDE_NAV_WIDTH,
  boxSizing: "border-box",
  fontSize: "14px",
  flexDirection: "column",
});

export const StyledCollectionFeedWrapper = glamorous.div({
  display: "flex",
  flexDirection: "column",
  boxSizing: "border-box",
  width: "100%",
  marginTop: 5,
});

export const StyledCollectionFeedItem = glamorous.div(
  {
    width: "100%",
    display: "flex",
    alignItems: "center",
    boxSizing: "border-box",
    padding: "5px 0 5px 35px",
    fontSize: 13,
    color: HALF_TRANSPARENT,
    transition: "all 100ms",
    ":hover": {
      cursor: "pointer",
      transition: "all 100ms",
      backgroundColor: "#303944",
    },
  },
  props => ({
    backgroundColor: props.active ? "#303944" : "auto",
  })
);

export const CollectionFeedImgWrapper = glamorous(FeedImgContainer)({
  marginRight: 10,
  width: 20,
  height: 20,
  border: "none",
  backgroundColor: "rgba(255, 255, 255, 0.9)",
  borderRadius: "50%",
});

export const CollectionFeedImg = glamorous.img({
  maxHeight: "100%",
  maxWidth: "100%",
  borderRadius: "50%",
});

export const StyledCollectionListItemTitle = glamorous.div(
  {
    padding: "5px 15px",
    display: "flex",
    alignItems: "center",
    fontSize: 14,
    color: SLIGHT_WHITE,
    height: 20,
    transition: "all 100ms",
    ":hover": {
      cursor: "pointer",
      transition: "all 100ms",
      backgroundColor: "#303944",
    },
  },
  props => ({
    backgroundColor: props.active ? "#303944" : "auto",
  })
);
