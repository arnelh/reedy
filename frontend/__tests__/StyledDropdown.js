import React from "react";
import { render } from "enzyme";
import { StyledDropdown } from "../styles/dropdown";

test("the component renders with default state ", () => {
  const wrapper = render(<StyledDropdown />);
  expect(wrapper).toMatchSnapshotWithGlamor();
});
