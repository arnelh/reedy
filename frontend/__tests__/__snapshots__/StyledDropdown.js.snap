// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the component renders with default state  1`] = `
.css-12cyyv6,
[data-css-12cyyv6] {
  background-color: #fff;
  position: fixed;
  height: 100px;
  top: -webkit-calc(55px + 3px);
  top: -moz-calc(55px + 3px);
  top: calc(55px + 3px);
  right: 0;
  width: 140px;
  border: 1px solid #dddede;
  border-radius: 4px;
  font-family: Oxygen;
  box-shadow: 0 3px 12px rgba(27,31,35,0.15);
  display: none;
}

.css-12cyyv6::before,
[data-css-12cyyv6]::before {
  content: close-quote;
  position: absolute;
  display: inline-block;
  top: -16px;
  right: 9px;
  left: auto;
  border: 8px solid transparent;
  border-bottom-color: #dddede;
}

.css-12cyyv6::after,
[data-css-12cyyv6]::after {
  content: close-quote;
  border: 7px solid transparent;
  border-bottom-color: #fff;
  position: absolute;
  display: inline-block;
  top: -14px;
  right: 10px;
  left: auto;
}

<div
  class="css-12cyyv6"
/>
`;
